src/app-layout/visual-refresh/context.tsx

```diff
@@ -399,7 +401,14 @@ export const AppLayoutInternalsProvider = React.forwardRef(
       drawersMaxWidth,
     });
 
+    const previousActiveDrawerId = useRef(activeDrawerId);
+
+    useEffect(() => {
+      previousActiveDrawerId.current = activeDrawerId;
+    }, [activeDrawerId]);
+
     const handleDrawersClick = (id: string | undefined, skipFocusControl?: boolean) => {
+      setPrevActiveDrawerId(previousActiveDrawerId.current);
```

Comment:
This is not guaranteed to be always the previous. For example, opening the drawer programmatically via "info" link does not visit this line, so `prevActiveDrawerId` won't be updated.

Following the comment from above I would recommend to avoid relying on previous value that much. Open for more discussion in DM